### DISCUSSION TITLE
How to get the currently selected configuration and platform

### DISCUSSION THREAD

**Discussion by ghost** [VsixCommunity/Community.VisualStudio.Toolkit#373] [Q&A]
Looking to get these currently selected values. Usually you'd go through the DTE > Solution > SolutionBuild > ActiveConfiguration. Can anyone help in pointing me towards how to get this through the `VS` static class, or other way, without the DTE

![image](https://user-images.githubusercontent.com/76792853/189660268-468f8143-c3b6-414e-9317-bd9166eddb49.png)

**Comment by reduckted** [comment]
If you're happy with a combined string value like `"Debug|AnyCPU"`, then you can use the `IVsSolutionBuildManager` service.

```cs
var manager = await VS.GetServiceAsync<SVsSolutionBuildManager, IVsSolutionBuildManager>();
var project = await VS.Solutions.GetActiveProjectAsync();

project.GetItemInfo(out var hierarchy, out _, out _);

IVsProjectCfg[] configs = new IVsProjectCfg[1];
ErrorHandler.ThrowOnFailure(manager.FindActiveProjectCfg(IntPtr.Zero, IntPtr.Zero, hierarchy, configs));

configs[0].getCanonicalName(out string name);
Debug.WriteLine(name); // Writes "Debug|AnyCPU" or "Release|x64", etc.
```

**Reply by ghost** [reply]
Thanks for the info! That’s a lot of code!
Do you think perhaps we should look into adding something into the toolkit to make this more accessible (so you could get or set it easily)?
